### ISSUE TITLE
Conflict with php open_basedir configuration

### ISSUE THREAD

**Issue by pgdavidApodis** [Nexmo/vonage-php-nexmo-bridge#1]
When COMPOSER_VENDOR_DIR is not set (it is not by default), the package generates a php warning, if php open_basedir configuration is set. 
```
Warning: file_exists(): open_basedir restriction in effect. File(/autoload.php) is not within the allowed path(s): (/xxx) in /yyy/vendor/vonage/nexmo-bridge/src/Autoloader.php on line 69
```

Existance of COMPOSER_VENDOR_DIR should be checked first in src/Autoload.php

```
private static function getClassLoader() : ClassLoader
{
if (file_exists(getenv('COMPOSER_VENDOR_DIR') . '/autoload.php')) {
    return include getenv('COMPOSER_VENDOR_DIR') . '/autoload.php';
}
if (file_exists(__DIR__ . '/../../../autoload.php')) {
    return include __DIR__ . '/../../../autoload.php';
}
if (file_exists(__DIR__ . '/../vendor/autoload.php')) {
    return include __DIR__ . '/../vendor/autoload.php';
}

throw new RuntimeException('Cannot detect composer autoload. Please run composer install');
}
```

First check should be replaced with :
```
if (getenv('COMPOSER_VENDOR_DIR') && file_exists(getenv('COMPOSER_VENDOR_DIR') . '/autoload.php'))
```

**Comment by natsumework** [comment]
I meet the same problem.

**Comment by sca1235** [comment]
Same issue

**Comment by KRens** [comment]
+1  Same issue.
Your solution fixes it, someone please merge this in.

**Comment by amirsefati** [comment]
Change to

```
if (file_exists(getenv('COMPOSER_VENDOR_DIR') . **__DIR__** . '/autoload.php')) {
            return include getenv('COMPOSER_VENDOR_DIR') . **__DIR__** . '/autoload.php';
        }
```

**Comment by nawidhaidari** [comment]
Hello KRens,
how did you users/change this to get it working.
I have this error **AH01071: Got error 'PHP message: PHP Warning: file_exists(): open_basedir restriction in effect. File(/autoload.php) is not within the allowed path(s): (/var/www/vhosts/MySite.com/:/tmp/) in /var/www/vhosts/MySite.com/httpdocs/vendor/vonage/nexmo-bridge/src/Autoloader.php on line 69', referer: https://mysite.com/**
and dont know how to get this fixed if you could please help.

Thank you,

**Comment by KRens** [comment]
THe first post written by pgdavidApodis contains the solution already.
Just replace 1 line of code in src/Autoload.php:

`if (file_exists(getenv('COMPOSER_VENDOR_DIR') . '/autoload.php')) {`
with:
`if (getenv('COMPOSER_VENDOR_DIR') && file_exists(getenv('COMPOSER_VENDOR_DIR') . '/autoload.php')) {`

**Comment by nawidhaidari** [comment]
Thank you @KRens. It did the trick. I added it to the autoloader.php because there were no if conditions in autoload.php.

**Comment by LucaRed** [comment]
This has been fixed by #2, so the issue can be closed.
